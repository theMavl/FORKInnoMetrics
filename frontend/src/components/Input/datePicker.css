.input {
    width: calc(var(--cell-size) * 1.5);

    outline: none;
    border: none;
    border-bottom: solid black 1px;
    text-align: center;
    cursor: pointer;
}