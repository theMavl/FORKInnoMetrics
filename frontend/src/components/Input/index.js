import moment from 'moment'
import React from 'react'
import PropTypes from 'prop-types'
import _ from 'lodash'
import DatePicker from 'react-datepicker'
// import Autosuggest from 'react-autosuggest'
import styles from './style.css'
import datePickerStyles from './datePicker.css'
// import autoSuggestStyles from './autoSuggest.css'

import 'react-datepicker/dist/react-datepicker-cssmodules.css'

class Input extends React.Component {
  render() {
    const {input, meta, type, label, labelStyle, required, error, containerProps, inputProps, autoSuggestProps, ...otherInputProps} = this.props
    let _containerProps = containerProps
    let _inputProps = otherInputProps
    _.assign(_inputProps, inputProps)
    let labelProps = {style: {}}
    _.assign(labelProps.style, labelStyle)
    if(_inputProps.id != undefined) {
      labelProps.htmlFor = _inputProps.id
    }
    _.assign(_inputProps, input)
    const displayedError = meta != undefined ? meta.error : error


    if(type === 'checkbox')
      return (
        <div className={styles.checkboxContainer} {..._containerProps}>
          <input className={styles.checkboxInput} {..._inputProps} type={type}/>
          <label className={styles.checkboxLabel} {...labelProps}>
            {label}
          </label>
        </div>
      )

    return (
        <div className={styles.container} {..._containerProps}>
          {label.trim().length > 0 && (
            <label className={styles.label} {...labelProps}>
              {label}
              {required && <span className={styles.required}>*</span>}
            </label>
          )}
          {type === 'datePicker' ?
           <DatePicker {...input}
                       onBlur={(value) => {input.onBlur(moment(input.value, 'DD/MM/YYYY').toDate())}}
                       dateFormat='dd/MM/yyyy'
                       popperPlacement='bottom-start'
                       popperModifiers={{
                         preventOverflow: {
                           enabled: true,
                           escapeWithReference: false, // force popper to stay in viewport (even when input is scrolled out of view)
                           boundariesElement: 'viewport'
                         }
                       }}
                       className={datePickerStyles.input}
           />
           // : type === 'autoSuggest' ?
           //   <Autosuggest {...autoSuggestProps} theme={autoSuggestStyles}
           //                inputProps={{className: styles.input, ..._inputProps}}
           //   />
           : <input className={styles.input} {..._inputProps} type={type}/>
          }
          <div className={styles.messages}>
            {_.get(meta, 'touched') && <span className={styles.error}>{displayedError}</span>}
          </div>
        </div>

    )
  }
}

Input.propTypes = {
  input: PropTypes.object,
  meta: PropTypes.object,
  type: PropTypes.string,
  label: PropTypes.string,
  required: PropTypes.bool,
  onChange: PropTypes.func,
  error: PropTypes.string
}

Input.defaultProps = {
  type: 'text',
  label: '',
  required: false,
  error: ''
}

export default Input