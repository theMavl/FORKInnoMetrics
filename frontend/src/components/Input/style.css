.container {
    width: 100%;
    height: calc(var(--cell-size) * 1.5);
    padding: 5px 0;
}

.label {
    display: block;
    width: 100%;
    margin: 0 0 5px 0;
}

.required {
    color: rgba(255, 0, 0, 1);
}

.input {
    display: block;
    width: 100%;
    height: calc(var(--cell-size) / 2);
    margin: 5px 0 5px 0;
    padding: 0 5px;
    border: none;
    border-radius: 5px;
    box-shadow: 0 0 5px 0px rgba(100, 100, 100, 0.15);
    border-bottom: 1px solid rgba(100, 100, 100, 0.25);
}

.input:focus {
    outline: none;
}

.messages {
    display: block;
    width: 100%;
    font-size: smaller;
}

.error {
    color: rgba(255, 0, 0, 1);
}



.checkboxContainer {
    composes: container;
    height: calc(var(--cell-size)*0.75);
    display: flex;
    align-items: center;
}

.checkboxLabel {
    composes: label;
    padding-left: 5px;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
}

.checkboxInput {
    composes: input;
    width: 24px;
    height: 24px;
    flex-shrink: 0;
}