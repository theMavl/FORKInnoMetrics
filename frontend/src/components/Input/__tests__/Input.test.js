import React from 'react'
import TestRenderer from 'react-test-renderer'
import Input from '../index'

describe('Test Input component', () => {

  it('Test render Input component', () => {

    const instance = TestRenderer.create(
      <Input/>
    )
    expect(instance.toJSON()).toMatchSnapshot()
  })
})