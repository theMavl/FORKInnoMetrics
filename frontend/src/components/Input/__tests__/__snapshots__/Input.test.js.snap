// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Input component Test render Input component 1`] = `
<div>
  <input
    type="text"
  />
  <div />
</div>
`;
